pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -697,9 +745,12 @@ impl RemoteTimelineClient {
             stopped.deleted_at = SetDeletedFlagProgress::InProgress(deleted_at);
 
             let mut index_part = IndexPart::new(
-                stopped.latest_files.clone(),
-                stopped.last_uploaded_consistent_lsn,
+                stopped.upload_queue_for_deletion.latest_files.clone(),
                 stopped
+                    .upload_queue_for_deletion
+                    .last_uploaded_consistent_lsn,
+                stopped
```

Comment:
why not have IndexPart::newFromUploadQueue and pass the upload_queue_for_deletion so it can extract whatever it needs from it ?

Reply:
Done. Though cannot simply refactor it to be used in `schedule_index_upload` because one place relies on this method being infallible, and metadata serialization is potentially fallible.